enumerables.rb

```diff
@@ -0,0 +1,201 @@
+# rubocop:disable
+module Enumerable
+  # my_each_Enumberable method
+  def my_each
+    return to_enum unless block_given?
+
+    arr = to_a
+    i = 0
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index Enumberable method
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    index = 0
+    my_each do |item|
+      yield(item, index)
+      index += 1
+    end
+
+    self
+  end
+
+  # my_select Enumberable method
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+
+    my_each { |num| array << num if yield(num) }
+
+    array
+  end
+
+  # my_all Enumberable method
+  def my_all?(argument = nil)
+    if argument
+      my_each { |element| return false unless argument === element } # rubocop:disable Style/CaseEquality
+    elsif block_given?
+      my_each { |element| return false unless yield(element) }
+    else
+      my_each { |element| return false unless element }
+    end
+    true
+  end
+
+  # my_any Enumberable method
+  def my_any?(argue = nil)
+    result = false
+    if block_given?
+      my_each { |val| return true if yield(val) }
+    end
+    if argue
+      my_each { |val| return true if argue === val } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |val| return true unless val }
+    end
+    result
+  end
+
+  # my_none Enumberable method
+  def my_none?(argu = nil)
+    result = true
+    if block_given?
+      my_each { |val| return false if yield(val) }
+    end
+    if argu
+      my_each { |val| return false if argu === val } # rubocop:disable Style/CaseEquality
+    else
+      my_each { |val| return false if val == true }
+    end
+    result
+  end
+
+  # my_count Enumberable method
+  def my_count(para = nil)
```

Comment:
`my_range` should also return the number of items when a range is passed.
It does not currently handle this well

Reply:
![Screenshot from 2021-01-08 14-19-10](https://user-images.githubusercontent.com/31193823/104050167-9460b780-51bc-11eb-8091-63bfe6c81116.png)